packages/cli/src/runners/task-runner-module.ts

```diff
@@ -92,4 +112,13 @@ export class TaskRunnerModule {
 			Container.get(InternalTaskRunnerDisconnectAnalyzer),
 		);
 	}
+
+	private onRunnerRestartLoopDetected = async (error: Error) => {
+		this.logger.error(error.message);
+		ErrorReporterProxy.error(error);
+
+		// Allow some time for the error to be flushed
+		await sleep(1000);
+		process.exit(1);
```

Comment:
Wondering if we could detect the task causing the restarts, trace the task back to the workflow, and deactivate it instead?

Reply:
The main purpose of this is to prevent an infinite loop if restarts in case the task runner process fails to start for whatever reason (e.g. a bug or some very abnormal condition that cause the process to exit before the runner even connects).

It would definitely be beneficial to have that type of logic. That's something we can consider in the future